test/test_site/testUtil/diffHtml.js

```diff
@@ -84,7 +86,9 @@ const diffHtml = (expected, actual) => {
     insidePath
     && startsWithClosedPath(fragment);
 
-  diff.forEach((part) => {
+  let error = false;
+  const diffPrinter = new DiffPrinter(filePathName);
+  diff.forEach((part, index) => {
```

Comment:
I did some extensive testing.

The new algorithm that uses `diffPrinter.printDiff()` seems to rely on the fact that the indices are sorted, but is that always truly the case? I was at one point able to find a case whereby this assumption seems to be violated, as it printed out two different parts of the file without the "...". I can't reproduce this bug again though, I don't know how to properly trigger it. :/